lib/fs/storage.go

```diff
@@ -91,6 +92,32 @@ func (s Storage) WalkWithContext(_ context.Context, path string, callback func(p
 	return s.Walk(path, callback)
 }
 
+// Copy copies a file.
+func (s Storage) Copy(src string, dst string, options ...map[string]interface{}) error {
+	mode := 0644
+
+	if len(options) > 0 {
+		if options[0]["mode"] != nil {
```

Comment:
We can omit `if` statements, and just a for loop here. Will be little bit less conditions. Also we would probably do a lookup in a the map, making sure that it's `nil` is not the same thing as checking if key in the map exists. Please double check type casting as well to omit errors. Something like:
```go
for _, opt := range options {
	if v, ok := opt["mode"]; ok {
		if m, ok := v.(int); ok {
			mode = m
		}
	}
}
```